### PR TITLE
Fix rcl package's logging API error specs and handling.

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. Just a few minor details I came across while writing tests.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `rcl`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11951/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6993/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9743/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11893/)

File: rcl/src/rcl/logging.c [hunk 1]
```diff
@@ -60,7 +60,8 @@ rcl_logging_configure_with_output_handler(
   rcl_logging_output_handler_t output_handler)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(global_args, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_ARGUMENT_FOR_NULL(allocator, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
```
Comment (brawner) [inline rcl/src/rcl/logging.c:63]: Why only set one of these to have an error message?

Comment (hidmic) [reply rcl/src/rcl/logging.c:63]: `RCL_CHECK_ARGUMENT_FOR_NULL` has built-in error messages (see [`RCUTILS_CHECK_ARGUMENT_FOR_NULL`](https://github.com/ros2/rcutils/blob/ed1f549b02a2054597aa339cbf529ab1e2151baf/include/rcutils/error_handling.h#L172)), but  `RCL_CHECK_ALLOCATOR` doesn't.

Comment (brawner) [reply rcl/src/rcl/logging.c:63]: Sounds good

Comment (Blast545) [reply rcl/src/rcl/logging.c:63]: I think that's the common scenario in the other files across `rcl`, `NULL` checks only return `RCL_RET_INVALID_ARGUMENT`

**Comment by hidmic** [comment]
Hmm [that PR job](http://build.ros2.org/job/Rpr__rcl__ubuntu_focal_amd64/203/testReport/junit/projectroot/test/test_get_node_names__rmw_fastrtps_cpp/) is reporting tests failures as if the ROS graph was dirty (dangling nodes, concurrent execution with other tests). A fluke? 

@ros-pull-request-builder retest this please.

**Comment by hidmic** [comment]
Uh, indeed.

**Pull request merged**